aggregator/src/main/scala/com/stratio/sparkta/aggregator/DataCube.scala

```diff
@@ -13,29 +13,51 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
 package com.stratio.sparkta.aggregator
 
+import java.io.{Serializable => JSerializable}
+
 import com.stratio.sparkta.sdk._
 import org.apache.spark.streaming.dstream.DStream
-import java.io.{Serializable => JSerializable}
 
+
+/**
+ * It builds a pre-calculated DataCube with dimension/s, rollup/s and operation/s defined by the user in the policy.
+ * Steps:
+ *   From a event stream it builds a Seq[(Seq[DimensionValue],Map[String, JSerializable])] with all needed data.
+ *   For each rollup it calculates aggregations taking the stream calculated in the previous step.
+ *   Finally, it returns a modified stream with pre-calculated data encapsulated in a UpdateMetricOperation.
+ *   This final stream will be used mainly by outputs.
+ * @param dimensions that will be contain the fields of the datacube.
+ * @param rollups that will be contain how the data will be aggregate.
+ */
 case class DataCube(dimensions: Seq[Dimension], rollups: Seq[Rollup]) {
 
+  /**
+   * It builds the DataCube calculating aggregations.
+   * @param inputStream with the original stream of data.
+   * @return the built DataCube.
+   */
   def setUp(inputStream: DStream[Event]): Seq[DStream[UpdateMetricOperation]] = {
-
-    //TODO: add event values
-    val extractedDimensionsStream: DStream[(Seq[DimensionValue], Map[String, JSerializable])] = inputStream
-      .map((e: Event) => {
-        val dimVals = for {
-          dimension: Dimension <- dimensions
-          value <- e.keyMap.get(dimension.name).toSeq
-          (bucketType, bucketedValue) <- dimension.bucketer.bucket(value)
-        } yield DimensionValue(dimension, bucketType, bucketedValue)
-        (dimVals, e.keyMap)
-      })
-      .cache()
-    // Create rollups
+    val extractedDimensionsStream = extractDimensionsStream(inputStream)
     rollups.map(_.aggregate(extractedDimensionsStream))
   }
 
-}
+  /**
+   * Extract a modified stream that will be needed to calculate aggregations.
+   * @param inputStream with the original stream of data.
+   * @return a modified stream after join dimensions, rollups and operations.
+   */
+  def extractDimensionsStream(inputStream: DStream[Event]):
```

Comment:
you're using extractDimensionsStream only in this class, maybe it couldn't be public